src/app/modules/main/profile_section/ens_usernames/controller.nim

```diff
@@ -63,44 +63,47 @@ proc init*(self: Controller) =
       return
     self.delegate.onUserAuthenticated(args.password)
 
+proc getNetwork*(self: Controller): NetworkDto =
+  return self.networkService.getNetworkForEns()
+
 proc checkEnsUsernameAvailability*(self: Controller, desiredEnsUsername: string, statusDomain: bool) =
   self.ensService.checkEnsUsernameAvailability(desiredEnsUsername, statusDomain)
 
-proc getMyPendingEnsUsernames*(self: Controller): seq[string] =
+proc getMyPendingEnsUsernames*(self: Controller): seq[EnsUsernameDto] =
   return self.ensService.getMyPendingEnsUsernames()
 
-proc getAllMyEnsUsernames*(self: Controller, includePendingEnsUsernames: bool): seq[string] =
+proc getAllMyEnsUsernames*(self: Controller, includePendingEnsUsernames: bool): seq[EnsUsernameDto] =
   return self.ensService.getAllMyEnsUsernames(includePendingEnsUsernames)
 
-proc fetchDetailsForEnsUsername*(self: Controller, ensUsername: string) =
-  self.ensService.fetchDetailsForEnsUsername(ensUsername)
+proc fetchDetailsForEnsUsername*(self: Controller, chainId: int, ensUsername: string) =
+  self.ensService.fetchDetailsForEnsUsername(chainId, ensUsername)
 
-proc setPubKeyGasEstimate*(self: Controller, ensUsername: string, address: string): int =
-  return self.ensService.setPubKeyGasEstimate(ensUsername, address)
+proc setPubKeyGasEstimate*(self: Controller, chainId: int, ensUsername: string, address: string): int =
+  return self.ensService.setPubKeyGasEstimate(chainId, ensUsername, address)
 
-proc setPubKey*(self: Controller, ensUsername: string, address: string, gas: string, gasPrice: string,
+proc setPubKey*(self: Controller, chainId: int, ensUsername: string, address: string, gas: string, gasPrice: string,
   maxPriorityFeePerGas: string, maxFeePerGas: string, password: string, eip1559Enabled: bool): string =
-  return self.ensService.setPubKey(ensUsername, address, gas, gasPrice, maxPriorityFeePerGas, maxFeePerGas, password, eip1559Enabled)
+  return self.ensService.setPubKey(chainId, ensUsername, address, gas, gasPrice, maxPriorityFeePerGas, maxFeePerGas, password, eip1559Enabled)
```

Comment:
why do u need to pass the chain ID down to the service while it could be resolved there? *question valid for most of the function here

Reply:
I think you're right, this is not a bad idea